src/javascript/components/auto-scroll.tsx

```diff
@@ -53,7 +53,8 @@ export const AutoScrollProvider: React.FC<{
   // and
   const observeScrollableNode = React.useCallback(
     (node: HTMLElement) => {
-      if (!didNav && navTarget === node.id) {
+      // TODO: replace with localeCompare
```

Comment:
Can you just do this now?

Reply:
I guess I didn't really get the intent of the TODO across, which is to replace it across the board (that is, have a matchesFragment function rather than have the logic spread in a dozen places)